spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/PositionDeletesRowReader.java

```diff
@@ -47,18 +47,20 @@ class PositionDeletesRowReader extends BaseRowReader<PositionDeletesScanTask>
   PositionDeletesRowReader(SparkInputPartition partition) {
     this(
         partition.table(),
+        partition.branch(),
         partition.taskGroup(),
         partition.expectedSchema(),
         partition.isCaseSensitive());
   }
 
   PositionDeletesRowReader(
       Table table,
+      String branch,
       ScanTaskGroup<PositionDeletesScanTask> taskGroup,
       Schema expectedSchema,
       boolean caseSensitive) {
 
-    super(table, taskGroup, expectedSchema, caseSensitive);
+    super(table, branch, taskGroup, expectedSchema, caseSensitive);
```

Comment:
All these changes make we wonder if we could hide the branching logic behind a special `Table` impl.